hack/generator/pkg/astmodel/type.go

```diff
@@ -63,11 +67,22 @@ type DeclarationContext struct {
 	Validations []KubeBuilderValidation
 }
 
+func CombineOverrides(overrides ...EqualityOverrides) EqualityOverrides {
```

Comment:
This isn't a combine - it's _use-the-last-one-provided-and-ignore-the-others_.

At minimum, needs a more accurate name ... better would be to have a composite override that really does combine.

Reply:
There’s no way that a Combine can work in general; some uses want to make equality stricter, some want to make it looser. I’ll rename this…

Reply:
… actually removing as YAGNI.